Makefile

```diff
@@ -56,7 +56,8 @@ TOOLS_DIR := $(PROJECT_DIR)/hack/internal/tools
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 BASE_IMAGE ?= gcr.io/distroless/static:nonroot
-BUILDER_IMAGE ?= golang:$(GO_VERSION)
+BASE_BUILDER_IMAGE ?= golang
```

Comment:
I would propose to change only this variable on test-infra without version.

@mimowo WDYT?

Reply:
Maybe, but I wouldn't couple this decision. 

Also, technically we may not want to use the "minimal version to compile" (as declared in go.mod.

For some reasons we may prefer to use a patched version, for example, garbage collector fix which improves performance in 1.26.3, so we may prefer using that, but still allow compiling for 1.26.0.

Reply:
Ah, we already extract the version from the go.mod. I guess it still makes sense as a fallback. If we ever have a case as described above we can introduce an overwrite.